src/Haskell/Ide/Engine/Transport/JsonStdio.hs

```diff
@@ -34,8 +36,8 @@ jsonStdioTransport cin = do
           putStr $ show (HieError (A.String $ T.pack $ show err))
           loop (cid + 1) stream'
         Just (Right r) -> do
-          writeChan cin (wireToChannel cout cid r)
-          rsp <- readChan cout
+          atomically $ writeTChan cin (wireToChannel cout cid r)
```

Comment:
I have not touched the transport side, pending the STX PR.

Once it is done, this will have to be split into two processes, one which reads on stdio and writes to the Chan to the dispatcher, the other waits on the return Chan and relays on to stdout.

But better not to mix two things at once, it can wait.